library/src/common/context/Select/index.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { ISelectContext } from '../../../types';
+
+const SelectContext = React.createContext<ISelectContext>({});
+
+const SelectProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
```

Comment:
When using TS, is it a good practice to always type hooks?

Reply:
Type all the things!

Entirely up to the dev and how big the code base is going to turn out, but better safe than sorry imo.